SNAPSHOTS.md

```diff
@@ -0,0 +1,201 @@
+
+   ```
+   apiVersion: v1
+   kind: PersistentVolumeClaim
+   metadata:
+      name: pvcfrmsnap1
+      namespace: default
+   spec:
+      accessModes:
+      - ReadWriteMany
+      resources:
+         requests:
+            storage: 1Gi
+      storageClassName: scfilesetinode
+      dataSource:
+         name: snap1
+         kind: VolumeSnapshot
+         apiGroup: snapshot.storage.k8s.io
+    
+   ```
+
+When creating a volume from a volume snapshot, data from source snapshot is copied to the newly created volume. This copy operation uses multiple Spectrum Scale nodes using mmapplypolicy. By default nodes on which the source filesystem is mounted are used. Users can define their own nodeclass to control the nodes where this copy operation should run based on their current workloads. Nodeclass must be defined on Spectrum Scale (ref. mmcrnodeclass). It can then be specified in the storageclass being used for creating PVC using parameter-
+
+   ```
+   nodeClass: <nodeclass_name>
+   ```
+
+### Static VolumeSnapshot
+You can expose a pre-existing Spectrum Scale fileset snapshot in Kubernetes by manually creating a VolumeSnapshotContent as below-
+
+   ```
+   apiVersion: snapshot.storage.k8s.io/v1
+   kind: VolumeSnapshotContent
+   metadata:
+      name: mysnapcontent
+   spec:
+      deletionPolicy: Delete
+      driver: spectrumscale.csi.ibm.com
+      source:
+         snapshotHandle: 18133600329030594550;0A1501E9:5F02F150;pvc-79e82f45-1b25-4d83-8ed0-c0d370cad5bd;mysnap
+      volumeSnapshotRef:
+         name: mysnapshot
+         namespace: default
+   ```
+
+Here snapshotHandle is of the format "clusterID;filesystem_UUID;filesetname;snapshotname;relative_path" where "relative_path" is optional.
+volumeSnapshotRef is the pointer to the VolumeSnapshot object this content should bind to.
+
+Once the VolumeSnapshotContent is created, create the VolumeSnapshot pointing to the same VolumeSnapshotContent
+
+   ```
+   apiVersion: snapshot.storage.k8s.io/v1
+   kind: VolumeSnapshot
+   metadata:
+      name: mysnapshot
+      namespace: default
+   spec:
+      source:
+         volumeSnapshotContentName: mysnapcontent
+   ```
+
+The VolumeSnapshot listing shows as below-
+
+   ```
+   # kubectl get volumesnapshot
+   NAME          READYTOUSE   SOURCEPVC   SOURCESNAPSHOTCONTENT   RESTORESIZE   SNAPSHOTCLASS   SNAPSHOTCONTENT                                    CREATIONTIME   AGE
+   mysnapshot    true                     mysnapcontent           0                             mysnapcontent                                      73m            72m
+   ```
+
+Note that here SOURCEPVC is empty and SOURCESNAPSHOTCONTENT points to the VolumeSnapshotContent. This snapshot can be normally used as a source while creating a PVC.
+
+Convenience script tools/generate_volsnapcontent_yaml.sh can be used to generate the volumeSnapshotContent yaml file.
+
+   ```
+   Usage: ./generate_volsnapcontent_yaml.sh
```

Comment:
github link changes with branches. For a readme this should be fine, when we fold this info in KC we can use github links.

Reply:
We can use link for dev branch or as you said we can use link in KC

Reply:
if we really wants to mention here we can use relative link also.

Reply:
Relative path is already given - tools/generate_volsnapcontent_yaml.sh. This is a readme that user is reading from github link so a path relative to readme makes perfect sense. You can find "tools" directory at the same level as snapshots.md. I dont think this will cause any confusion, but a broken link can. Also this readme is relevant only during development phase, post GA we will be putting all this information in KC and just point to KC from here (just like the main readme) or remove this readme altogether.

Reply:
Reason I was mentioning was, till the time we have kc ready not everyone will be aware of from where to download
https://raw.githubusercontent.com/IBM/ibm-spectrum-scale-csi/dev/tools/generate_volsnapcontent_yaml.sh

But I am ok this the current way